core/include/scipp/core/element/unary_operations.h

```diff
@@ -111,13 +111,14 @@ constexpr auto negative_inf_to_num_out_arg =
 constexpr auto reciprocal = overloaded{
     arg_list<double, float>,
     [](const auto &x) noexcept {
-      return static_cast<
-                 core::detail::element_type_t<std::decay_t<decltype(x)>>>(1) /
-             x;
-    },
-    [](const units::Unit &unit) {
-      return units::Unit(units::dimensionless) / unit;
-    }};
+        return static_cast<
+                   core::detail::element_type_t<std::decay_t<decltype(x)>>>(1) /
+               x;
+} // namespace element
```

Comment:
I had trouble with this before, I believe it is a bug in clang-format: If you remove the "noexcept" a couple lines above it should work as normally. Since this is inline anyway it should be safe to remove the noexcept to avoid this, I believe?

Reply:
I found only one situation where `noexcept` might make a difference:

```
constexpr auto f1 = []() noexcept {};
constexpr auto f2 = [](){};
void foo(void (*f)() noexcept)
{
  f();
}
void bar(void (*f)())
{
  f();
}
int main()
{
  foo(f1);
  // foo(f2);  // error: invalid user-defined conversion
  bar(f1);
  bar(f2);
  return 0;
}
```

But I don't think it's so significant. Anyway, initially I used clang-format 10.0.0 and it doesn't have this bug. If/when it is possible, you can try to upgrade the tool.